modules/nlu/src/backend/election/natural-election.test.ts

```diff
@@ -0,0 +1,48 @@
+import { extractElectedIntentSlot } from './natural-election'
+import { fakeEventUnderstanding } from '../../test/types'
+
+describe('naturalElection', () => {
+  describe('extractElectedIntentSlot', () => {
+    it("should return input directly if the 'predictions' property is nil", () => {
+      const input = fakeEventUnderstanding({ predictions: undefined })
+      const result = extractElectedIntentSlot(input)
+      expect(result).toStrictEqual(input)
+    })
+
+    it("should return input directly if the 'intent' property is nil", () => {
+      const input = fakeEventUnderstanding({ predictions: {}, intent: undefined })
+      const result = extractElectedIntentSlot(input)
+      expect(result).toStrictEqual(input)
+    })
+
+    it("should return input directly if the 'context' property of the 'predictions' property is nil", () => {
+      const input = fakeEventUnderstanding({ predictions: { context: undefined }, intent: {} })
+      const result = extractElectedIntentSlot(input)
+      expect(result).toStrictEqual(input)
+    })
+
+    it("should return input directly if the 'intents' property of the prediction 'context' property is nil", () => {
+      const input = fakeEventUnderstanding({ predictions: { context: { intents: undefined } }, intent: {} })
+      const result = extractElectedIntentSlot(input)
+      expect(result).toStrictEqual(input)
+    })
+
+    it("should return input with an empty 'slots' object if no labels within the 'intents' array match the 'intent' name", () => {
+      const input = fakeEventUnderstanding({
+        predictions: { context: { intents: [{ label: 'name', confidence: 1, extractor: 'extractor', slots: [] }] } },
+        intent: { name: 'label' }
+      })
+      const result = extractElectedIntentSlot(input)
+      expect(result).toStrictEqual({ ...input, slots: {} })
+    })
+
+    it("should return input with the 'slots' object filled with that of the elected intent if a label within the 'intents' array matches the 'intent' name", () => {
```

Comment:
would probably be a good idea to test with actual slots to see if they appear in the result object ?

Reply:
Done